### ISSUE TITLE
crystal spec needs -lxml2?

### ISSUE THREAD

**Issue by amitsaha** [crystal-lang/crystal#3262]
Consider the code  [here](https://github.com/amitsaha/linux_voice_6/tree/master/broadcaster). If I run `crystal spec` here, I get this:

```
$ crystal spec
/usr/bin/ld: cannot find -lxml2
collect2: error: ld returned 1 exit status
Error: execution of command failed with code: 1: `cc -o "/home/asaha/.cache/crystal/crystal-run-spec.tmp" "${@}"  -rdynamic  -lz `command -v pkg-config > /dev/null && pkg-config --libs libssl || printf %s '-lssl -lcrypto'` `command -v pkg-config > /dev/null && pkg-config --libs libcrypto || printf %s '-lcrypto'` -lxml2 -lpcre -lm -lgc -lpthread /opt/crystal/src/ext/libcrystal.a -levent -lrt -ldl -L/usr/lib -L/usr/local/lib`
```

**Comment by kirbyfan64** [comment]
Try installing the libxml package for your repo.

**Comment by amitsaha** [comment]
Yeah, that's what I did. But, shouldn't the crystal package when installed via a RPM package specify it as a dependency?

**Comment by RX14** [comment]
I think the packages should depend on stuff like libxml2-devel and openssl-devel, which aren't technically required dependencies, but are required to do any serious work in crystal. This - in conjunction with a better error message when missing libraries - would definitely help newcomers to crystal.

**Comment by amitsaha** [comment]
@RX14 +1, I also raised the issue of the dependency on linker in https://github.com/crystal-lang/crystal/issues/3207

**Comment by straight-shoota** [comment]
The immediate issue is fixed by #6004, `spec` no longer depends on `libxml`.

But it would still make sense to think about adding dev packages as dependencies.
